### PR TITLE
Improve README documentation

### LINKED ISSUE #49: Improve README.md documentation

### PR DESCRIPTION
**Pull request by aricasas**
This PR will add to and improve our README documentation, mostly concerning feedback received from Milestone 4 from our TA

### CODE REVIEW

File: README.md
Comment (katarinaavucic) [inline README.md:150]: We should include the MVC pattern since our TA pointed it out in our rubric!

File: README.md
Comment (katarinaavucic) [inline README.md]: Also I think we should write more for these, because this is where we are supposed to be explaining our choices! Something like:

"The Factory design pattern is used to implement the Add/Edit Category and Add/Edit Expense features. We chose to add this implementation to avoid tight coupling between the creators (ExpenseUseCaseInteractor and CategoryUseCaseInteractor) and their products (Expense and Category). This choice helps our code adhere to the Open/Closed Principle, since we can introduce new types of MonthObjects into the code and avoid breaking our current implementation. Additionally, it adheres to the Single Responsibility Principle, as the product creation code is moved to a separate class, allowing the UseCaseInteractors to focus on error handling and post-creation tasks."

Feel free to add or modify!

**Review by katarinaavucic — CHANGES_REQUESTED** [review]
It looks super good and very professional! My only comment is on the design pattern section! :)

File: README.md
Comment (aricasas) [reply README.md]: Ohh, thank you for pointing this out. I integrated this paragraph now!

File: README.md
Comment (aricasas) [reply README.md:150]: Oooh, yep. I forgot that one also counts as a design pattern

Comment (aricasas) [reply README.md:150]: Does anyone remember why we ended choosing MVC over MVP? Cause I'd like to add our reasoning, but I don't think I remember our discussions about it

Comment (katarinaavucic) [reply README.md:150]: Hmm I remember it was Yin's idea, but I am pretty sure it was just because it was suggested in the slides. Maybe we could mention how it helps structure our code in a way that adheres to Clean Architecture?

**Comment by katarinaavucic** [comment]
I just realized that we will need to redo the Javadocs and Code Coverage reports after Lulu pushes her PR.

File: README.md
Comment (aricasas) [reply README.md:150]: Ok. I just added a new commit that contains MVC. Could you check it to see if I missed something?

Comment (katarinaavucic) [reply README.md:150]: Beautiful!

**Review by katarinaavucic — COMMENTED** [review]
I will approve once we update the Javadoc and coverage report (I have added myself to the coverage issue so I will do that as soon as Lulu's PR is merged)!

**Comment by aricasas** [comment]
Sounds good! For this one, the only thing that needs to change is the test coverage screenshot, so we only need to wait for that PR

File: README.md
Comment (YinLingithub) [inline README.md:123]: Maybe rename this to Implementation and also move the Testing and Javadoc sections to here? I feel like these two sections are not really a part of Usages. 

The Implementation section below could be renamed to Implementation Design or Implementation Structure.

Comment (aricasas) [reply README.md:123]: Oh that's a good idea. I just made an Implementation section with Code Structure (formerly Implementation), Testing, and JavaDoc as its subsections

Comment (YinLingithub) [reply README.md:123]: Yep! Look great!

**Review by YinLingithub — DISMISSED** [review]
Looks great!

**Review by katarinaavucic — DISMISSED** [review]
Looks perfect!!!

**Review by YinLingithub — DISMISSED** [review]
Looks great!

**Review by luluwuwu — DISMISSED** [review]
great!

**Review by katarinaavucic — APPROVED** [review]
Yay!

**Review by luluwuwu — APPROVED** [review]
approved

**Review by YinLingithub — APPROVED** [review]
Approving again.

**Pull request merged**
